internal/command/deploy/machines.go

```diff
@@ -179,7 +175,14 @@ func NewMachineDeployment(ctx context.Context, args MachineDeploymentArgs) (Mach
 		onlyRegions:            args.OnlyRegions,
 		immediateMaxConcurrent: immedateMaxConcurrent,
 		volumeInitialSize:      volumeInitialSize,
+		processGroups:          args.ProcessGroups,
+	}
+	// TODO: Blend extraInfo into ValidationError and remove this hack
+	if err, extraInfo := md.Validate(ctx); err != nil {
+		fmt.Fprintf(iostreams.FromContext(ctx).ErrOut, extraInfo)
+		return nil, err
```

Comment:
why was this block moved?

Reply:
The call to `md.Validate()` needed to be moved after the `md` object was defined, but I moved it back in 31a8581a04d9209dbcb5536247f6c9837fc24c51 which changes the call to `appConfig.ValidateGroups(ctx, lo.Keys(args.ProcessGroups))`.